### PR TITLE
Requirable factories #2 

### PR DESCRIPTION
**Pull request by mamhoff**
This PR builds on #623 (and includes @Sinetheta s work). I went through all factories, set up tests that don't require `testing_support/factories`, and made sure all factories `build` and `create`. As the factories are IMO code, and a pretty important part of the framework, I suggest to leave the tests in the repo. They can later be used to improve the factories even more. I do understand they seem pretty bogus at times, but having a systematic way of testing factories is a nice thing.

### CODE REVIEW

**Comment by jhawthorn** [comment]
Isn't setting `ENV['NO_FACTORIES'] = "NO FACTORIES"` going to cause test run order issues? If one of the factory specs is loaded before the other specs in our system I fear all factories will not be loaded, though they should be. I don't think "other factories not loaded" is something we can test for.

Otherwise I love the factory specs.

**Comment by cbrunsdon** [comment]
I'm also not really excited about the env check, I know it means we have a test to make sure that factories would be individually loadable, but I'm not sure thats worth the potential extra issues by controlling loading with ENV settings. Will think about it.

Tests on our factories if fantastic though, thanks for the effort. The huge reliance on our factories by ourselves and other projects without any reasonable suite has always been a black mark on the project IMO.

**Comment by mamhoff** [comment]
The `ENV` check was necessary for testing the requireability of the factories. I entirely understand the ugliness of it. 

I'll remove it - for knowing whether any factories still break, we could do one of two things: 
1. Just wait for the community to cry when a factory does not load individually (probably only happens for a couple of popular factories (`order`, `shipment`, `payment`, `line_item`)
2. Remove loading all spree factories from the spec helper and load factories in individual spec files. That'd be a large amount of work though, and probably concern of another PR. 

If anyone has a better idea on how to test `require`ability ... I'd be excited.

**Comment by cbrunsdon** [comment]
We could look to do it outside of our specs as well just as a CI task. 

While individually requirable is nice, I'd definitely say not worth confusing up our specs. I'd even say we try waiting on backlash for a while. We'll also be watching PR's to make sure we're not adding dependencies to factories without including them, so we might be optimising for a problem we won't have.

**Comment by mamhoff** [comment]
Switch removed :)

**Comment by cbrunsdon** [comment]
As this is just going into Kevin's original PR I am :+1:

**Comment by cbrunsdon** [comment]
Whoops, sorry, missed where it was being targeted it to.

File: core/lib/spree/testing_support/sequences.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+require 'factory_girl'
```
Comment (jhawthorn) [inline core/lib/spree/testing_support/sequences.rb]: I think this file would be better under `testing_support/factories` (but I don't feel that strongly about it)

**Comment by jhawthorn** [comment]
:+1: Regardless of my suggestion. This is great.

File: core/lib/spree/testing_support/sequences.rb (hunk 1 above)
Comment (mamhoff) [inline core/lib/spree/testing_support/sequences.rb]: I wouldn't mind doing that, but that was Kevin's decision, and I didn't want to touch his stuff. @Sinetheta should the file be moved?

**Comment by jordan-brough** [comment]
:+1: thanks!  Would you mind rebasing and resolving the merge conflicts?  Looks like we're ready to merge after that.

**Comment by mamhoff** [comment]
@jordan-brough rebased!

**Comment by jordan-brough** [comment]
Perfect, thanks again!

**Pull request merged**